Update Sonata getting started docs. (#250)
microsoft/cheriot-rtos: docs/GettingStarted.md

# apt install xmake
```

### Installing u2futils

> If you are working with Sonata, you will need to convert the ELF files that the linker produces to [USB Flashing Format (UF2)](https://github.com/microsoft/uf2) format.

Author: USB Flashing Format format?

When you use a simulator, the UART output goes directly to the simulator's standard output.
In contrast, when you run on an FPGA the output will go to a USB serial device.

You will need to connect a terminal emulator to the corresponding serial port.
The Sonata system exposes *four* serial devices, typically numbered 0-3.
The UART is device 2.
On Linux, this is usually `/dev/ttyUSB2` (if no other USB serial devices are connected).
On macOS, the device name includes the serial number of the device and so will be something like `/dev/tty.usbserial-LN28292` (the last digit is the serial port number).
On Windows, it will be a COM port, but unfortunately Windows is not always consistent about the order in which the ports appear.

If you install a serial terminal (for example, `minicom` on *NIX or PuTTY on Windows), you can point it at the relevant serial port.

The serial setup is the same as for the Arty A7 [above](#installing-and-running-the-firmware), aside from the different device name.

Once a serial terminal is connected, running `xmake run` should show the output from the test suite.

> If you are unsure which of Sonata's serial ports is the correct one, you can connect your serial console to all four and see which produces output from the test suite.

Reviewer: I don't think it's a good idea to connect to all of them at the same time. I think one might be the debugger and actually halts CPU or something.
```suggestion
If you are unsure which of Sonata's serial ports is the correct one, you can connect your serial console to each of them in turn and see which produces output from the test suite.
```

Author: > I don't think it's a good idea to connect to all of them at the same time. I think one might be the debugger and actually halts CPU or something.

I connected to all four on my Mac to figure out the right one and it was fine.
